<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/pygment_trac.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <title>Super-deduper by dstreett</title>
  </head>

  <body>
    <header>
      <div class="inner">
        <h1>Super-deduper</h1>
        <h2>An application to remove PCR duplicates from high throughput sequencing runs.</h2>
        <a href="https://github.com/dstreett/Super-Deduper" class="button"><small>View project on</small> GitHub</a>
      </div>
    </header>

    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h3>
<a id="super-duper" class="anchor" href="#super-deduper" aria-hidden="true"><span class="octicon octicon-link"></span></a>Introducing the Super Deduper</h3>

<p>The Super Deduper is a C/C++ application that was created to remove PCR duplicates from high throughput sequencing data. First I will give an overview on installation then go over how to run it and how it works. If you run into any problems or have any questions, please, submit an issue on the github page. I hope you find this application useful in your research. </p>

<p>Here is the basic installation and basic use case of the Super Deduper. It has been tested on Linux as well as OS X. </p>

<pre><code>$ git clone git@github.com:dstreett/Super-Deduper.git
$ cd Super-Deduper
$ make
$ echo "Example of Paired End inputs (Two files [-1 and -2] and Interleaved [-i])"
$ ./super_deduper -1 ./examples/phix_example_PE1 -2 ./examples/phix_example_PE2
$ ./super_deduper -i ./examples/interleaved_example.fastq.gz
$ echo "Example of Single End Reads [-U]"
$ ./super_deduper -U ./examples/phix_example_SE.fastq.gz
</code></pre>


<h3>
<a id="designer-templates" class="anchor" href="#designer-templates" aria-hidden="true"><span class="octicon octicon-link"></span></a>How Super Deduper Works</h3>

<p>
The Super Deduper turns a region of the sequence (or the entire sequence) into a binary number. It does this by mapping A to 00, T to 11, C to 01, and G to 10.
Super deduper concatenates read 1's and read 2's binary numbers together and creates a unique ID for the pair that is then used to determine duplications in a binary search tree. With single end reads the same thing happens, except there is no read 2 to concatenate with.
Both the read and the reverse compliment are tested for - however, only the larger of the two are kept. Doing this ensures that the reverse compliment is tested for and that all reads that are duplicates (be in a reverse compliment duplicate) are tested and mapped to the same unique ID.
A node is created in this tree if a new sequence ID is seen. If an ID is all ready seen a summation of the quality score of the entire read is taken. The read with the higher summation is taken.
The reason for a summation versus an average is summations give preferential treatment to longer reads as well as higher quality scores.
</p>

<h3>
<a id="arguements" class="anchor" href="#arguemnts" aria-hidden="true"><span class="octicon octicon-link"></span></a>Arguements and what they do</h3>

<p>
Here are the basic arguements
</p>


<p>If you are doing paired end reads (that aren't interleaved)</p>
<pre><code>-1, --read1 [path to read 1]
-2, --read2 [path to read 2]
</code></pre>

<p>
If you want to use an interleaved file - no problem
</p>
<pre><code>-i, --interleaved-input [path to interleaved file]</code></pre>


<p>Additionally, single ends are handled just fine</p>

<pre>
<code>-U, --singleend [path to single end file] </code>
</pre>

<p><b>All these options can be comma seperated to add multiple files. Additionally, fastq or fastq.gz are accepted as input.</b></p>

<p>To change the region in which the Super Deduper determines duplicates here are the arguments</p>

<pre><code>-s, -start NUM [starting base pair of ID]
-l, --length NUM [number of base pairs from the starting base pair]
</pre></code>

<p><b>If you enter inputs that are out of bounds for reads the application will stop running and exit.</b><p>


<p>Output arguments are as follows</p>

<pre><code>-p, --output-prefix PREFIX [This will add PREFIX to the start of "_nodup_PE1.fastq]
-o, --interleaved-output PREFIX [Outputs in interleaved format with prefix specified]
-g, --gzip-output [Outputs in gzipped format]
</code></pre>


<p>If you would like to save the duplication sceme for a later data, you can actually output the tree and input it the next time you run it.</p>
<pre><code>-O, --output-tree PATH [outputs tree to the file]
-I, --input-tree PATH [Tree files to read in (comma seperated)]
</code></pre>


<h3>
<a id="authors-and-contributors" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span class="octicon octicon-link"></span></a>Thank You</h3>

<p>I would like to thank everyone who took the time to download and use this application. In efforts to continuely import and grow, let me know any issues you run into or if anything is unclear.</p>

<h3>

        </section>

        <aside id="sidebar">
          <a href="https://github.com/dstreett/Super-Deduper/zipball/master" class="button">
            .zip file
          </a>
          <a href="https://github.com/dstreett/Super-Deduper/tarball/master" class="button">
            <small>Download</small>
            .tar.gz file
          </a>

          <p class="repo-owner"><a href="https://github.com/dstreett/Super-Deduper"></a> is maintained by <a href="https://github.com/dstreett">dstreett</a>.</p>

          <p>This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the Architect theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.</p>
        </aside>
      </div>
    </div>

  
  </body>
</html>
